diesel_codegen/src/embed_migrations.rs

```diff
@@ -0,0 +1,91 @@
+use syn;
+use quote;
+
+use diesel::migrations::{migration_paths_in_directory, version_from_path};
+use diesel_codegen_shared::migration_directory_from_given_path;
+use std::error::Error;
+use std::path::Path;
+
+use util::{get_options_from_input, get_option};
+
+pub fn derive_embed_migrations(input: syn::MacroInput) -> quote::Tokens {
+    fn bug() -> ! {
+        panic!("This is a bug. Please open a Github issue \
+               with your invocation of `embed_migrations!");
+    }
+
+    let options = get_options_from_input(&input.attrs, bug);
+    let migrations_path_opt = options.map(|o| get_option(&o, "migrations_path", bug));
+    let migrations_expr = migration_directory_from_given_path(migrations_path_opt)
+        .and_then(|path| migration_literals_from_path(&path));
+    let migrations_expr = match migrations_expr {
+        Ok(v) => v,
+        Err(e) => panic!("Error reading migrations: {}", e),
```

Comment:
Isn't that `match` the same as `migrations_expr.expect("Error reading migrations")` (also print the error afterwards)?